client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -122,10 +146,16 @@ private void handleMissingLog() {
      * message find which new segment it should go to and send it there. This can happen recursively if segments turn
      * over very quickly.
      */
-    @GuardedBy("lock")
     private void handleLogSealed(Segment segment) {
-        List<PendingEvent> toResend = selector.refreshSegmentEventWritersUponSealed(segment);
-        resend(toResend);
+        //TODO: shrids since this is executed by the callback for that segment we need to pass the events to be resent.
+        lock.readLock().lock();
+        try {
+            List<PendingEvent> toResend = selector.refreshSegmentEventWritersUponSealed(segment,
+                    segmentSealedCallBack);
+            resend(toResend);
```

Comment:
Updated the code to handle this scenario. Kindly review.
It ensures the following
- Prevents parallel `sealedsegmentsCallbacks ` from being executed.
- It ensures no writes happen during the meantime thereby ensuring the order of the events.